akka-stream/src/main/scala/akka/stream/impl/io/OutputStreamSourceStage.scala

```diff
@@ -172,29 +108,16 @@ private[akka] class OutputStreamAdapter(
   private[this] def sendData(data: ByteString): Unit =
     send(() ⇒ {
       try {
-        dataQueue.put(data)
+        // FIXME take into account writeTimeout here.
```

Comment:
fixme, isn't that just a `tryAcquire(permits, timeout, unit)`?

Reply:
It is, I hadn't noticed `tryAcquire` there :smile: